web/src/peer_data.ts

```diff
@@ -1,16 +1,47 @@
+import {z} from "zod";
+
 import * as blueslip from "./blueslip.ts";
+import * as channel from "./channel.ts";
 import {LazySet} from "./lazy_set.ts";
+import {page_params} from "./page_params.ts";
 import type {User} from "./people.ts";
 import * as people from "./people.ts";
 import * as sub_store from "./sub_store.ts";
 
 // This maps a stream_id to a LazySet of user_ids who are subscribed.
 const stream_subscribers = new Map<number, LazySet>();
+const fetched_stream_ids = new Set<number>();
```

Comment:
I think you want to split this up. Let's do a prep PR that adds this field and populates it during `peer_data.initialize` with the channels that had a `subscribers` field, and processes a `partial_subscribers` field if present.

Reply:
#34282